### DISCUSSION TITLE
how to do "git push --force --mirror" with go-git

### DISCUSSION THREAD

**Discussion by liruixin** [go-git/go-git#1055] [Q&A]
how to do "git push --force --mirror" with go-git

**Comment by liruixin** [comment]
`	opts := &git.PushOptions{
		RemoteName: remote.Config().Name,
		Auth:       sshKey,
		Force:      true,
		RefSpecs:   []config.RefSpec{"+refs/*:refs/*"},
	}`

m not sure about using this push options : "RefSpecs:   []config.RefSpec{"+refs/*:refs/*"}" is ok

**Comment by pjbgf** [comment]
That is correct, `--force` and `--mirror` in `git` is the same as the push options below:

```golang
    pushOpts := &git.PushOptions{
        RefSpecs: []config.RefSpec{
            config.RefSpec("+refs/*:refs/*"),
        },
        Force: true,
    }
    
    err = remote.PushContext(context.Background(), pushOpts)
    if err != nil {
        return err 
    }
```
